File: app/src/main/java/com/dailyvery/apps/imhome/BatteryReceiver.java
package com.dailyvery.apps.imhome;

import android.app.Notification;
import android.app.NotificationManager;
import android.app.PendingIntent;
import android.content.BroadcastReceiver;
import android.content.Context;
import android.content.Intent;
import android.telephony.SmsManager;
import android.widget.Toast;

import com.dailyvery.apps.imhome.Data.Avert;
import com.dailyvery.apps.imhome.Data.AvertDataSource;

import java.sql.SQLException;
import java.util.List;

/**
 * Created by Utilisateur on 23/01/2017.
 */

public class BatteryReceiver extends BroadcastReceiver {

    //TODO Factoriser createNotification qui est utilise dans tous les receiver
    private final void createNotification(Context context, String message, int notifID){
        NotificationManager notificationManager = (NotificationManager) context.getSystemService(context.NOTIFICATION_SERVICE);

        // prepare intent which is triggered if the
        // notification is selected

        Intent intent = new Intent(context, MainActivity.class);
        // use System.currentTimeMillis() to have a unique ID for the pending intent
        PendingIntent pIntent = PendingIntent.getActivity(context, (int) System.currentTimeMillis(), intent, 0);

        // build notification
        // the addAction re-use the same intent to keep the dailyvery short
        Notification n  = new Notification.Builder(context)
                .setContentTitle(context.getString(R.string.notifName))
                .setContentText(message)
                .setSmallIcon(R.drawable.ic_done_white_24dp)
                .setContentIntent(pIntent)
                .setAutoCancel(true)
                .getNotification();

        notificationManager.notify(notifID, n);
    }
    @Override
    public void onReceive(Context context, Intent intent) {
        AvertDataSource ads = new AvertDataSource(context);
        try {
            ads.open();
            List<Avert> avertList = ads.getAllAvert();
            if (avertList.size() > 0) {
                int notifID = avertList.get(0).getHashcode();
                for (Avert a : avertList) {
                    SmsManager.getDefault().sendTextMessage(a.getContactNumber(), null, context.getString(R.string.messageCriticalBattery), null, null);
                    Toast.makeText(context, context.getString(R.string.notifMessageCriticalBattery) + " " + a.getContactName() + " " + context.getString(R.string.recipientAlerted), Toast.LENGTH_LONG).show();
                    createNotification(context, context.getString(R.string.notifMessageCriticalBattery) + " " + a.getContactName(), notifID++);
                }
            }
        } catch (SQLException e) {
            e.printStackTrace();
        } finally {
            ads.close();
        }
    }
}

File: app/src/main/java/com/dailyvery/apps/imhome/BatteryReceiver.kt
package com.dailyvery.apps.imhome

import android.app.Notification
import android.app.NotificationManager
import android.app.PendingIntent
import android.content.BroadcastReceiver
import android.content.Context
import android.content.Intent
import android.telephony.SmsManager
import android.widget.Toast

import com.dailyvery.apps.imhome.Data.Avert
import com.dailyvery.apps.imhome.Data.AvertDataSource

import java.sql.SQLException

/**
 * Created by Utilisateur on 23/01/2017.
 */

class BatteryReceiver : BroadcastReceiver() {

    //TODO Factoriser createNotification qui est utilise dans tous les receiver
    private fun createNotification(context: Context, message: String, notifID: Int) {
        val notificationManager = context.getSystemService(context.NOTIFICATION_SERVICE) as NotificationManager

        // prepare intent which is triggered if the
        // notification is selected

        val intent = Intent(context, MainActivity::class.java)
        // use System.currentTimeMillis() to have a unique ID for the pending intent
        val pIntent = PendingIntent.getActivity(context, System.currentTimeMillis().toInt(), intent, 0)

        // build notification
        // the addAction re-use the same intent to keep the dailyvery short
        val n = Notification.Builder(context)
                .setContentTitle(context.getString(R.string.notifName))
                .setContentText(message)
                .setSmallIcon(R.drawable.ic_done_white_24dp)
                .setContentIntent(pIntent)
                .setAutoCancel(true)
                .notification

        notificationManager.notify(notifID, n)
    }

    override fun onReceive(context: Context, intent: Intent) {
        val ads = AvertDataSource(context)
        try {
            ads.open()
            val avertList = ads.allAvert
            if (avertList.size > 0) {
                var notifID = avertList[0].hashcode
                for (a in avertList) {
                    SmsManager.getDefault().sendTextMessage(a.contactNumber, null, context.getString(R.string.messageCriticalBattery), null, null)
                    Toast.makeText(context, context.getString(R.string.notifMessageCriticalBattery) + " " + a.contactName + " " + context.getString(R.string.recipientAlerted), Toast.LENGTH_LONG).show()
                    createNotification(context, context.getString(R.string.notifMessageCriticalBattery) + " " + a.contactName, notifID++)
                }
            }
        } catch (e: SQLException) {
            e.printStackTrace()
        } finally {
            ads.close()
        }
    }
}
